scripts/spark_wdl/ReadsPipelineSparkMulticore.wdl

```diff
@@ -0,0 +1,70 @@
+workflow ReadsPipelineSparkWorkflow {
+  File ref_fasta
+  File ref_fasta_index
+  File? ref_fasta_gzi
+  File ref_dict
+  File input_bam
+  String output_vcf_basename
+  File known_sites
+  File known_sites_index
+
+  String gatk_docker
+  String gatk
+  Int? num_cpu_override
+  Int num_cpu = select_first([num_cpu_override, 16])
+
+  call ReadsPipelineSpark {
+    input:
+      ref_fasta = ref_fasta,
+      ref_fasta_index = ref_fasta_index,
+      ref_fasta_gzi = ref_fasta_gzi,
+      ref_dict = ref_dict,
+      input_bam = input_bam,
+      output_vcf_basename = output_vcf_basename,
+      known_sites = known_sites,
+      known_sites_index = known_sites_index,
+      gatk_docker = gatk_docker,
+      gatk = gatk,
+      num_cpu = num_cpu
+  }
+
+  output {
+    File output_vcf = ReadsPipelineSpark.output_vcf
+  }
+}
+
+task ReadsPipelineSpark {
+  File ref_fasta
+  File ref_fasta_index
+  File? ref_fasta_gzi
+  File ref_dict
+  File input_bam
+  String output_vcf_basename
+  File known_sites
+  File known_sites_index
+
+  String gatk_docker
+  String gatk
+  Int num_cpu
+
+  command {
+    ${gatk} \
+      ReadsPipelineSpark \
+      -R ${ref_fasta} \
+      -I ${input_bam} \
+      -O ${output_vcf_basename}.vcf \
+      --known-sites ${known_sites} \
+      -pairHMM AVX_LOGLESS_CACHING \
+      --max-reads-per-alignment-start 10
+  }
+
+  runtime {
+    docker: gatk_docker
+    cpu: num_cpu
+    disks: "local-disk 4000 SSD"
```

Comment:
Two comments, first that you probably want to actually use LOCAL SSDs (as the persistent SSDs are a very poor value proposition). This is done py specifying any number under 3TB for `LOCAL` disks. This will automatically be stitched together as a logical volume. Second we should probably make this vary with the size of the bam. I don't know if you have a good sense of what the disk space requirements are for the tool but it should decide on the disk size dynamically

Reply:
Here is an example of how to do that:
     ```Float agg_bam_size = size(bam_file, "GB")
     disk_size = (sort_disk_multiplier * agg_bam_size) + additional_disk```

Reply:
You're right about persistent SSDs - we don't really want to use those if possible. I've fixed the code to use `LOCAL`. I don't have a good sense of the overhead needed, so I've left the size as 3000GB, but made it configurable.

I've tested the new WDL on Terra.